### PR TITLE
feat(schematron): allow verification of Schematron text

### LINKED ISSUE #1634: Feature request: check content of Schematron message reported
It can be desirable in some cases to test the content of the SVRL `failed-assert|successful-report` message. 
This could be expressed as the content of the `x:expect-*` elements, e.g.

    <x:expect-assert id="unknown-pi">This PI is unknown.</x:expect-assert>

Dynamically-constructed messages complicate matters, i.e. where Schematron `assert|report` contain `value-of|name` - unless XSpec were to construct these messages too from the test case - which sounds possible, but excessive?

In these cases, it could be that the XSpec assertion about the content of the message is only applied where a single node is targeted (by `@location` or `@count`), as here, where the quoted text is constructed by  `value-of`:

    <x:expect-assert id="a0001"
                 location="/html[1]/body[1]/div[5]">Unexpected content: "Nosy nodes nobody knows"</x:expect-assert>

### PR DESCRIPTION
**Pull request by galtm**
This pull request adds support for users who want to use `<x:expect-assert>` or  `<x:expect-report>` to verify the messages of asserts and reports. The messages can be either the main message (content of `<sch:assert>` or `<sch:report>`) or diagnostics triggered by the `<sch:assert>` or `<sch:report>` element. Further details about the design are in my [comment](https://github.com/xspec/xspec/issues/1634#issuecomment-1837544364).

In my local workspace, I tested the change on Windows both on this branch and on a branch I created from the `schxslt` branch where I cherry-picked these commits.

Closes #1634

@AndrewSales , if you're able to try this out, let me know if it suits your needs.

### Follow-up

- [ ] Update wiki

### CODE REVIEW

File: src/schematron/schut-to-xspec.xsl [hunk 1]
```diff
@@ -232,6 +234,26 @@
...
     <xsl:template match="@count | @label | @pending" as="empty-sequence()" mode="make-predicate" />
 
+    <xsl:template match="x:expect-assert | x:expect-report" as="text()" mode="make-text-predicate">
+        <xsl:variable name="x-expect-text-space-normalized" as="xs:string"
+            select="normalize-space(.)"/>
+        <xsl:variable name="apos-char" as="xs:string">'</xsl:variable>
+        <xsl:variable name="x-expect-text-apos-adjusted" as="xs:string"
+            select="replace($x-expect-text-space-normalized, $apos-char, $apos-char || $apos-char)"/>
+        <xsl:variable name="x-expect-text-content-wrapped" as="xs:string"
+            select="$apos-char || $x-expect-text-apos-adjusted || $apos-char"/>
```
Comment (AirQuick) [inline src/schematron/schut-to-xspec.xsl:244]: Looks like these lines can be replaced by a call to `x:quote-with-apos()` defined in `src/common/common-utils.xsl`, or at least `$apos-char` can be replaced by `$x:apos`.

Comment (galtm) [reply src/schematron/schut-to-xspec.xsl]: Great idea. I improved this piece of code in 551d3a63c74e23c02916db9455fb0c3bf0799d70.

**Comment by galtm** [comment]
> [x] Update wiki

I added a paragraph [here](https://github.com/xspec/xspec/wiki/Writing-Scenarios-for-Schematron#xexpect-) ([diff](https://github.com/xspec/xspec/wiki/Writing-Scenarios-for-Schematron/_compare/564015c86b6c35cb2af81c91349a75045a7083a7...8fab17a38a0901db623de92a53dccfff29608d83)).

After we release XSpec v3.0, the wiki will need adjustment in the "Upcoming XSpec v3.0 feature" label -- either remove or rephrase the label.

**Comment by AndrewSales** [comment]
I've now had chance to test, @galtm, and it works as expected - many thanks for this.

**Pull request merged**